crates/core/src/contracts/erc20/ethereum_erc20.rs

```diff
@@ -49,4 +75,58 @@ impl<'a, P: Provider + Send + Sync> EthereumErc20<'a, P> {
             actual: balance.len(),
         })?)
     }
+
+    pub async fn name(self, block_id: BlockId) -> Result<String, EthApiError<P::Error>> {
+        // Prepare the calldata for the bytecode function call
+        let calldata = IERC20Calls::Symbol(SymbolCall).encode();
+        let calldata = calldata.into_iter().map(FieldElement::from).collect();
+
+        let block_id = EthBlockId::new(block_id);
+        let block_id: StarknetBlockId = block_id.try_into()?;
+
+        let result = self.kakarot_contract.eth_call(&self.address, calldata, &block_id).await?;
+        let name: Vec<u8> = result.0.into();
+
+        Ok(String::from_utf8(name.clone()).map_err(|_| DataDecodingError::InvalidReturnArrayLength {
+            entrypoint: "name".into(),
+            expected: 32,
+            actual: name.len(),
+        })?)
+    }
+
+    pub async fn symbol(self, block_id: BlockId) -> Result<String, EthApiError<P::Error>> {
+        // Prepare the calldata for the bytecode function call
+        let calldata = IERC20Calls::Symbol(SymbolCall).encode();
+        let calldata = calldata.into_iter().map(FieldElement::from).collect();
+
+        let block_id = EthBlockId::new(block_id);
+        let block_id: StarknetBlockId = block_id.try_into()?;
+
+        let result = self.kakarot_contract.eth_call(&self.address, calldata, &block_id).await?;
+        let symbol: Vec<u8> = result.0.into();
+
+        Ok(String::from_utf8(symbol.clone()).map_err(|_| DataDecodingError::InvalidReturnArrayLength {
+            entrypoint: "symbol".into(),
+            expected: 32,
```

Comment:
same for symbol, string is of arbitrary length

Reply:
see above, proposition to add another variant